lms/views/api/error.py

```diff
@@ -12,52 +13,125 @@
 _ = i18n.TranslationStringFactory(__package__)
 
 
-@exception_view_config(context=ValidationError, renderer="json")
-def validation_error(context, request):
-    request.response.status_int = 422
-    # For frontend requests to proxy API endpoints, handle schema
-    # validation errors.
-    return {"message": context.explanation, "details": context.messages}
+@view_defaults(renderer="json")
+class ExceptionViews:
+    """
+    Exception views for the API.
 
+    Error responses from the API have JSON bodies with the following keys (all
+    optional):
 
-@exception_view_config(context=CanvasAPIAccessTokenError, renderer="json")
-def canvas_api_access_token_error(request):
-    request.response.status_int = 400
-    # For a CanvasAPIAccessTokenError we don't send any error message or
-    # details to the frontend because we don't want the frontend to show any
-    # error message to the user in this case. Just the 400 status so the
-    # frontend knows that the request failed, and that it should show the user
-    # an [Authorize] button so they can get a (new) access token and try again.
-    return {"message": None, "details": None}
+    1. "auth_url": A URL to (re-)authenticate with the third-party API that's
```

Comment:
This key is not present in any of the responses currently, but I understand that you're going to add it shortly.

Reply:
I'll edit this out for now so that we can merge this. We don't know for sure yet whether we'll end up putting `"auth_url"` in the responses